metaserver/src/main/scala/scala/meta/languageserver/Main.scala

```diff
@@ -26,7 +26,11 @@ object Main extends LazyLogging {
       System.setErr(err)
       logger.info(s"Starting server in $workspace")
       logger.info(s"Classpath: ${Properties.javaClassPath}")
-      server.start()
+      val server = new ScalametaLanguageServer(cwd, out)
+      val launcher = LSPLauncher.createServerLauncher(server, System.in, out)
```

Comment:
A question not related to lsp4j itself, but more in the context of rewriting this:
Would it be hard to get rid of this mechanism of passing `cwd` through a var? I think that in LSP supposes that you can launch a server independently and then tell it where your project is through the initialization parameters (`rootUri`).

Reply:
Yes, I think that's the way to do it. I wanted to take on that, as soon as I get it up and running